pkg/github/client/client.go

```diff
@@ -53,28 +54,28 @@ var runnerPerMinuteRate = map[string]float64{
 }
 
 // New instantiates a new GitHub API client.
-func New(ctx context.Context, settings models.Settings) (*Client, error) {
+func New(ctx context.Context, settings models.Settings, opts httpclient.Options) (*Client, error) {
```

Comment:
Now that we have the context passed down we can read the http client options here instead of passing down from the datasource.go and instance.go